packages/google_maps_flutter/google_maps_flutter_web/lib/src/convert.dart

```diff
@@ -203,30 +203,25 @@ gmaps.InfoWindowOptions? _infoWindowOptionsFromMarker(Marker marker) {
 
   // Add an outer wrapper to the contents of the infowindow, we need it to listen
   // to click events...
-  final HtmlElement container = DivElement()
+  final HTMLElement container = createDivElement()
     ..id = 'gmaps-marker-${marker.markerId.value}-infowindow';
 
   if (markerTitle.isNotEmpty) {
-    final HtmlElement title = HeadingElement.h3()
+    final HTMLHeadingElement title = (document.createElement('h3') as HTMLHeadingElement)
       ..className = 'infowindow-title'
       ..innerText = markerTitle;
-    container.children.add(title);
+    container.appendChild(title);
   }
   if (markerSnippet.isNotEmpty) {
-    final HtmlElement snippet = DivElement()
+    final HTMLElement snippet = createDivElement()
       ..className = 'infowindow-snippet'
       // `sanitizeHtml` is used to clean the (potential) user input from (potential)
       // XSS attacks through the contents of the marker InfoWindow.
       // See: https://pub.dev/documentation/sanitize_html/latest/sanitize_html/sanitizeHtml.html
       // See: b/159137885, b/159598165
-      // The NodeTreeSanitizer.trusted just tells setInnerHtml to leave the output
-      // of `sanitizeHtml` untouched.
       // ignore: unsafe_html
-      ..setInnerHtml(
-        sanitizeHtml(markerSnippet),
-        treeSanitizer: NodeTreeSanitizer.trusted,
-      );
-    container.children.add(snippet);
+      ..innerHTML = sanitizeHtml(markerSnippet);
```

Comment:
There is no `NodeSanitizer` anymore, not sure of the implications.

Reply:
I think the "safe" way to solve this now is to :

* https://developer.mozilla.org/en-US/docs/Web/API/TrustedHTML

Then define a setter for `HTMLElement` that lets us set the `innerHTML` property with a `TrustedHTML` instance, rather than the current `Node` or `HTMLElement` type.

The `sanitizeHtml` method could be the implementation of the Trusted Type policy, like described in the MDN example:

```js
const escapeHTMLPolicy = trustedTypes.createPolicy("flutter-sanitize-maps", {
  createHTML: (String html) => sanitizeHtml(html), // Returns an instance of TrustedHTML
});

let el = document.getElementById("myDiv");
const escaped = escapeHTMLPolicy.createHTML(markerSnippet);
el.safeInnerHTML = escaped; // safeInnerHTML only accepts TrustedHTML objects.
```

Reply:
The support for `TrustedHTML` is a bit lacking though, neither Safari nor Firefox doesn't support it.

Reply:
> neither Safari nor Firefox support it.

And probably never will, but that's what the `unsafe_html` lint is actually looking at 😱